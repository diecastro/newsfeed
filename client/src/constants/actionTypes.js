export default {
  getNewsRequest: 'GET_NEWS_REQUEST',
  getNewsSuccess: 'GET_NEWS_SUCCESS',
  getNewsFailure: 'GET_NEWS_FAILURE',
  loginUserRequest: 'LOGIN_USER_REQUEST',
  loginUserSuccess: 'LOGIN_USER_SUCCESS',
  loginUserFailure: 'LOGIN_USER_FAILURE',
  getAuthorsRequest:'GET_AUTHORS_REQUEST',
  getAuthorsSuccess:'GET_AUTHORS_SUCCESS',
  getAuthorsFailure:'GET_AUTHORS_FAILURE',
  postAuthorRequest:'POST_AUTHOR_REQUEST',
  postAuthorSuccess:'POST_AUTHOR_SUCCESS',
  postAuthorFailure:'POST_AUTHOR_FAILURE',
  deleteNewsRequest:'DELETE_NEWS_REQUEST',
  deleteNewsSuccess:'DELETE_NEWS_SUCCESS',
  deleteNewsFailure:'DELETE_NEWS_FAILURE'
};
