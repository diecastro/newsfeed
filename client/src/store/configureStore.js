let configureStore;

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./configureStore.prod');
} else {
  configureStore = require('./configureStore.dev').default;
  module.exports = configureStore;
}
