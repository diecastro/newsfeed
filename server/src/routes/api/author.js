const express = require('express');
const router = express.Router();
const passport = require('passport');
const authorService = require('../../services/author');

router.post('/author', passport.authenticate('jwt', {session: false}), async (req, res, next) => {
  authorService.addAuthor(req.body).then(result => {
    res.json(result);
  }).catch(error => {
    res.json({message: `An error has ocurred, ${error.message}`})
  });
});
router.get('/authors', async (req, res, next) => {
  authorService.getAuthors().then(result => {
    res.json(result);
  }).catch(error => {
    res.json({message: `An error has ocurred, ${error.message}`})
  });
});

module.exports = router;
