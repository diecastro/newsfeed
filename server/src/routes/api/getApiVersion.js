

'use strict';

const config = require('../../config/app'),
  applicationVersion = require('./../../../../package.json');

module.exports = function (request, response) {
  return response.status(config.status.ok)
    .send(applicationVersion.version);

};
