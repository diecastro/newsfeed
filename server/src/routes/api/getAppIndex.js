

'use strict';

const config = require('../../config/app');

module.exports = function (request, response) {
  return response.status(config.status.ok)
    .send(config.server.name);
};
