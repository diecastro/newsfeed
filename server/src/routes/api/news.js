const express = require('express');
const router = express.Router();
const newsService = require('../../services/news');

router.post('/news', async (req, res, next) => {
  newsService.addNews(req.body).then(result => {
    res.json(result);
  }).catch(error => {
    res.json({message: `An error has ocurred, ${error.message}`})
  });
});

router.get('/news/:page', async (req, res, next) => {
  newsService.getNews(req).then(result => {
    res.json(result)
  }).catch(error => {
    res.json({message: `An error has ocurred, ${error.message}`})
  })
});

router.delete('/news', async (req, res, next) => {
  newsService.deleteNews(req.body).then(result => {
    res.json(result);
  }).catch(error => {
    res.json({message: `An error has ocurred, ${error.message}`})
  });
});

router.put('/news', async (req, res, next) => {
  newsService.updateNews(req.body).then(result => {
    res.json(result);
  }).catch(error => {
    res.json({message: `An error has ocurred, ${error.message}`})
  });
});

module.exports = router;
