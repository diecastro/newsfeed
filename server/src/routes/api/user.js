const passport = require('passport');
const express = require('express');
const router = express.Router();
const jwt = require('jsonwebtoken');
const config = require('../../config/app');

router.post('/login', async (req, res, next) => {
  passport.authenticate('userLogin', async (err, user, info) => {
    try {
      if (err || !user) {
        const error = new Error('An Error occurred');
        return res.json({message: error.message});
      }
      req.login(user, {session: false}, async (error) => {
        if (error) return next(error);
        const token = jwt.sign(user, config.server.secret);
        return res.json({token});
      });
    } catch (error) {
      return res.json({message: `An error has ocurred, ${error.message}`});
    }
  })(req, res, next);
});

router.post('/signup', passport.authenticate('userSignUp', {session: false}), async (req, res, next) => {
  res.json({
    message: 'User Created'
  });
});

module.exports = router;
