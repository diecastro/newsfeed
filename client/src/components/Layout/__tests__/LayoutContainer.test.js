import React from 'react';
import ConnectedLayoutContainer from '../LayoutContainer';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import actionTypes from '../../../constants/actionTypes';
import api from '../../../middleware/api';
import _ from 'lodash';

const middlewares = [thunk, api];
const mockStore = configureStore(middlewares);

describe('components/Layout/LayoutContainer', () => {

  let shoppingBag = {test: 123, finalized: false, accessories: {}, plan: {planCode: 0}};
  let stubs = {};

  let commonProperties = {
    title: 'myLayout',
    syncBag: jest.fn(),
    removeAllItemsFromBag: jest.fn(),
    clearNotification: jest.fn(),
    checkTokenExpiration: jest.fn(),
    toggleRefreshTokenDialog: jest.fn(),
    refreshToken: jest.fn(),
    logoutUser: jest.fn(),
    openDialog: jest.fn(),
    getProspect: jest.fn(),
    setProspectId: jest.fn(),
    toggleVisibilitySignInForm: jest.fn(),
    router: [],
    auth: {
      isAuthenticated: false,
      newToken: true
    }
  };

  let auth = {
    isAuthenticated: false,
    isFetching: false,
    loginError: null,
    menuActive: false,
    newToken: null,
    signupError: null,
    tokenAutoRenew: false,
    tokenTermsRenew: null
  };

  const startState = {
    userExperience: {
      title: 'myLayout',
      notifications: [],
      dispatch: stubs.dispatch,
      isUpdatingCart: false,
      cart: {
        productIds: []
      }
    },
    shoppingBag,
    auth,
    application: {
      threatMetrixTemplate: 'some text'
    }
  };

  let expectedActions =
    [{
      type: actionTypes.syncBagRequest
    }];

  let store, wrapper;

  beforeEach(() => {
    commonProperties.location = {query: {}, pathname: '/somePath'};
    commonProperties.children = <div className='someChild'/>;
  });

  afterEach(function () {
    _.invokeMap(stubs, 'restore');
  });

  describe('Connected LayoutContainer', () => {

    beforeEach(() => {
      store = mockStore({...startState});
      wrapper = shallow(<ConnectedLayoutContainer store={store} {...commonProperties}/>).shallow();
    });

    it('should render correct snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders the connected component', () => {
      expect(wrapper.find('.Layout').length).toEqual(1);
    });

    it('should not render backdrop class', () => {
      const startState = {
        userExperience: {
          title: 'myLayout',
          notifications: [],
          dispatch: stubs.dispatch,
          megamenu: true,
          isUpdatingCart: false
        },
        shoppingBag,
        auth,
        application: {
          threatMetrixTemplate: 'some text'
        }
      };
      store = mockStore({...startState});
      const wrapper = shallow(<ConnectedLayoutContainer store={store} {...commonProperties} />).shallow();
      expect(wrapper.find('.backdrop.hidden').length).toBeFalsy();
    });

    it('should render backdrop class', () => {
      const startState = {
        userExperience: {
          title: 'myLayout',
          notifications: [],
          dispatch: stubs.dispatch,
          isUpdatingCart: false
        },
        application: {status: 'finalized', threatMetrixTemplate: 'some text'},
        location: {query: {}, pathname: '/someOtherPath'},
        shoppingBag,
        auth
      };
      store = mockStore({...startState});
      const wrapper = shallow(<ConnectedLayoutContainer store={store} {...commonProperties} />).shallow();
      expect(wrapper.find('.backdrop.hidden').length).toBeTruthy();
    });

    describe('componentWillReceiveProps', () => {

      it('should complete', () => {
        let nextProps = {
          shoppingBag: {
            test: 456,
            finalized: false
          },
          userExperience: {
            isUpdatingCart: true
          },
          application: {status: 'finalized', threatMetrixTemplate: 'some text'},
          location: {query: {}, pathname: '/someOtherPath'},
          notifications: [{type: 'Success', message: 'someSuccess'}],
          auth
        };
        wrapper.instance().componentWillReceiveProps(nextProps);
        expect(store.getActions()).toEqual([
          {type: actionTypes.getThreatMetrixTagRequest},
          {type: actionTypes.syncBagRequest},
          {type: actionTypes.updateCartSuccess}
        ]);
      });

      it('should call syncBag if the shoppingBag changes', () => {
        let nextProps = {
          shoppingBag: {
            test: 456,
            finalized: false
          },
          userExperience: {
            isUpdatingCart: true
          },
          application: {status: 'finalized', threatMetrixTemplate: 'some text'},
          location: {query: {}, pathname: '/someOtherPath'},
          notifications: [{type: 'Success', message: 'someSuccess'}],
          auth
        };
        wrapper.instance().componentWillReceiveProps(nextProps);
        expect(store.getActions()).toEqual([
          {type: actionTypes.getThreatMetrixTagRequest},
          {type: actionTypes.syncBagRequest},
          {type: actionTypes.updateCartSuccess}
        ]);
      });

      it('should call syncBag if the shoppingBag changes', () => {
        let nextProps = {
          shoppingBag: {
            test: 456,
            finalized: false
          },
          userExperience: {
            isUpdatingCart: true
          },
          application: {status: '', threatMetrixTemplate: 'some text'},
          location: {query: {}, pathname: '/someOtherPath'},
          notifications: [{type: 'Success', message: 'someSuccess'}],
          auth
        };
        wrapper.instance().componentWillReceiveProps(nextProps);
        expect(store.getActions()).toEqual([
          {type: actionTypes.getThreatMetrixTagRequest},
          {type: actionTypes.syncBagRequest},
          {type: actionTypes.updateCartSuccess}
        ]);
      });

    });

    it('should render the footer component', () => {
      expect(wrapper.find('Footer').length).toEqual(1);
    });

    describe('mapDispatchToProps actions', () => {

      beforeEach(() => {
        commonProperties.location = {query: {}};
        store = mockStore({...startState});
        wrapper = shallow(<ConnectedLayoutContainer store={store} {...commonProperties} />);
      });

      it('calls syncBag action', () => {

        wrapper.props().syncBag({shoppingBag: 456});
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActions);
      });

      describe('clearApplicationStorage', () => {

        it('calls removeAllItemsFromBag action with clear', () => {

          wrapper.props().clearApplicationStorage();
          const actualActionTypes = store.getActions();
          expect(actualActionTypes).toEqual([
            {
              type: actionTypes.clearAllItems
            }
          ]);
        });

      });

      it('calls clearNotification action', () => {
        let expectedActions =
          [{
            notificationIndex: 1,
            type: actionTypes.clearNotification
          }];
        wrapper.props().clearNotification(1);
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActions);
      });

      it('calls checkTokenExpiration', () => {

        wrapper.props().checkTokenExpiration();
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual([]);
      });

      it('calls toggleRefreshTokenDialog', () => {
        const expectedActions = [
          {
            type: actionTypes.toggleRefreshTokenDialog,
            value: true
          }
        ];
        wrapper.props().toggleRefreshTokenDialog(true);
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActions);
      });

      it('calls refreshToken', () => {
        const expectedActions = [
          {
            type: actionTypes.refreshTokenRequest
          }
        ];
        wrapper.props().refreshToken();
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActions);
      });

      it('calls logoutUser', () => {
        const expectedActions = [
          {
            isAuthenticated: true,
            isFetching: true,
            type: actionTypes.logoutRequest
          },
          {
            type: actionTypes.expireTokenRequest
          },
          {
            isAuthenticated: false,
            isFetching: false,
            type: actionTypes.logoutSuccess
          }
        ];
        wrapper.props().logoutUser();
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActions);
      });

      it('calls openDialog', () => {
        const expectedActions = [
          {
            payload: {activeDialog: 'dialog', isDialogOpen: true},
            type: actionTypes.openDialog
          }
        ];
        wrapper.props().openDialog('dialog');
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActions);
      });

      it('calls toggle', () => {
        const expectedActions = [
          {type: 'TOGGLE_VISIBILITY_SIGNIN_FORM'}
        ];
        wrapper.props().toggleVisibilitySignInForm();
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActions);
      });

    });

    describe('refreshToken', () => {

      const expectedActions = [{type: actionTypes.getThreatMetrixTagRequest},
        {
          'type': 'REFRESH_TOKEN_REQUEST'
        }
      ];

      it('should call functions', () => {
        wrapper.instance().refreshToken();
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActions);
      });

      it('should call functions with token', () => {
        commonProperties.auth.newToken = true;
        wrapper.instance().refreshToken();
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActions);
      });

    });

    describe('handleTokenClose', () => {

      const expectedActionsNoToken = [
        {type: actionTypes.getThreatMetrixTagRequest},
        {
          'isAuthenticated': true,
          'isFetching': true,
          'type': 'LOGOUT_REQUEST'
        },
        {
          type: actionTypes.expireTokenRequest
        },
        {
          'isAuthenticated': false,
          'isFetching': false,
          'type': 'LOGOUT_SUCCESS'
        },
        {
          'payload': {
            'activeDialog': 'auth',
            'isDialogOpen': true
          },
          'type': 'DIALOG_OPEN'
        },
        {
          'type': 'TOGGLE_VISIBILITY_SIGNIN_FORM'
        }
      ];

      const expectedActionsToken = [
        {type: actionTypes.getThreatMetrixTagRequest},
        {
          'type': 'TOGGLE_REFRESH_TOKEN_DIALOG',
          'value': false
        },
        {
          'isAuthenticated': true,
          'isFetching': true,
          'type': 'LOGOUT_REQUEST'
        },
        {
          type: actionTypes.expireTokenRequest
        },
        {
          'isAuthenticated': false,
          'isFetching': false,
          'type': 'LOGOUT_SUCCESS'
        },
        {
          'payload':
            {
              'activeDialog': 'auth',
              'isDialogOpen': true
            },
          'type': 'DIALOG_OPEN'
        },
        {
          'type': 'TOGGLE_VISIBILITY_SIGNIN_FORM'
        }
      ];

      it('should call logoutUser', () => {
        wrapper.instance().handleTokenClose();
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActionsNoToken);
      });

      it('should call logoutUser with token', () => {
        localStorage.setItem('pa-apply-token', 'someToken');
        wrapper.instance().handleTokenClose();
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual(expectedActionsToken);
      });

    });

    describe('checkToken', () => {

      it('should call checkTokenExpiration', () => {
        wrapper.instance().checkToken(commonProperties);
        const actualActionTypes = store.getActions();
        expect(actualActionTypes).toEqual([{type: actionTypes.getThreatMetrixTagRequest}]);
      });

    });

  });

  describe('Prospect Id ppid', () => {

    beforeEach(() => {
      commonProperties.location = {query: {ppid: '777'}, search: '?ppid=777'};
      store = mockStore({...startState});
      wrapper = shallow(<ConnectedLayoutContainer store={store} {...commonProperties}/>).shallow();
    });

    it('should render correct snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders the connected component', () => {
      expect(wrapper.find('.Layout').length).toEqual(1);
    });

  });

});
