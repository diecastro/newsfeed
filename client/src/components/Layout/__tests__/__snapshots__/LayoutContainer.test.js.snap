// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout/LayoutContainer Connected LayoutContainer should render correct snapshot 1`] = `
<div
  className="Layout"
>
  <div
    className="backdrop hidden"
  />
  <Header
    application={
      Object {
        "threatMetrixTemplate": "some text",
      }
    }
    auth={
      Object {
        "isAuthenticated": false,
        "isFetching": false,
        "loginError": null,
        "menuActive": false,
        "newToken": null,
        "signupError": null,
        "tokenAutoRenew": false,
        "tokenTermsRenew": null,
      }
    }
    cartUpdateSuccess={[Function]}
    checkTokenExpiration={[Function]}
    clearApplicationStorage={[Function]}
    clearNotification={[Function]}
    getProspect={[Function]}
    getThreatMetrixTags={[Function]}
    location={
      Object {
        "pathname": "/somePath",
        "query": Object {},
      }
    }
    logoutUser={[Function]}
    notifications={Array []}
    openDialog={[Function]}
    performCartAction={[Function]}
    refreshToken={[Function]}
    removeAllItemsFromBag={[MockFunction]}
    router={Array []}
    setProspectId={[Function]}
    shoppingBag={
      Object {
        "accessories": Object {},
        "finalized": false,
        "plan": Object {
          "planCode": 0,
        },
        "test": 123,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    syncBag={[Function]}
    title="myLayout"
    toggleRefreshTokenDialog={[Function]}
    toggleVisibilitySignInForm={[Function]}
    userExperience={
      Object {
        "cart": Object {
          "productIds": Array [],
        },
        "dispatch": undefined,
        "isUpdatingCart": false,
        "notifications": Array [],
        "title": "myLayout",
      }
    }
  >
    <div
      className="someChild"
    />
  </Header>
  <div
    application={
      Object {
        "threatMetrixTemplate": "some text",
      }
    }
    auth={
      Object {
        "isAuthenticated": false,
        "isFetching": false,
        "loginError": null,
        "menuActive": false,
        "newToken": null,
        "signupError": null,
        "tokenAutoRenew": false,
        "tokenTermsRenew": null,
      }
    }
    cartUpdateSuccess={[Function]}
    checkTokenExpiration={[Function]}
    className="someChild"
    clearApplicationStorage={[Function]}
    clearNotification={[Function]}
    getProspect={[Function]}
    getThreatMetrixTags={[Function]}
    location={
      Object {
        "pathname": "/somePath",
        "query": Object {},
      }
    }
    logoutUser={[Function]}
    notifications={Array []}
    openDialog={[Function]}
    performCartAction={[Function]}
    refreshToken={[Function]}
    removeAllItemsFromBag={[MockFunction]}
    router={Array []}
    setProspectId={[Function]}
    shoppingBag={
      Object {
        "accessories": Object {},
        "finalized": false,
        "plan": Object {
          "planCode": 0,
        },
        "test": 123,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    syncBag={[Function]}
    title="myLayout"
    toggleRefreshTokenDialog={[Function]}
    toggleVisibilitySignInForm={[Function]}
    userExperience={
      Object {
        "cart": Object {
          "productIds": Array [],
        },
        "dispatch": undefined,
        "isUpdatingCart": false,
        "notifications": Array [],
        "title": "myLayout",
      }
    }
  >
    <div
      className="someChild"
    />
  </div>
  <Footer />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "some text",
      }
    }
  />
</div>
`;

exports[`components/Layout/LayoutContainer Prospect Id ppid should render correct snapshot 1`] = `
<div
  className="Layout"
>
  <div
    className="backdrop hidden"
  />
  <Header
    application={
      Object {
        "threatMetrixTemplate": "some text",
      }
    }
    auth={
      Object {
        "isAuthenticated": false,
        "isFetching": false,
        "loginError": null,
        "menuActive": false,
        "newToken": null,
        "signupError": null,
        "tokenAutoRenew": false,
        "tokenTermsRenew": null,
      }
    }
    cartUpdateSuccess={[Function]}
    checkTokenExpiration={[Function]}
    clearApplicationStorage={[Function]}
    clearNotification={[Function]}
    getProspect={[Function]}
    getThreatMetrixTags={[Function]}
    location={
      Object {
        "query": Object {
          "ppid": "777",
        },
        "search": "?ppid=777",
      }
    }
    logoutUser={[Function]}
    notifications={Array []}
    openDialog={[Function]}
    performCartAction={[Function]}
    refreshToken={[Function]}
    removeAllItemsFromBag={[MockFunction]}
    router={
      Array [
        "shopping-bag",
        "shopping-bag",
      ]
    }
    setProspectId={[Function]}
    shoppingBag={
      Object {
        "accessories": Object {},
        "finalized": false,
        "plan": Object {
          "planCode": 0,
        },
        "test": 123,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    syncBag={[Function]}
    title="myLayout"
    toggleRefreshTokenDialog={[Function]}
    toggleVisibilitySignInForm={[Function]}
    userExperience={
      Object {
        "cart": Object {
          "productIds": Array [],
        },
        "dispatch": undefined,
        "isUpdatingCart": false,
        "notifications": Array [],
        "title": "myLayout",
      }
    }
  >
    <div
      className="someChild"
    />
  </Header>
  <div
    application={
      Object {
        "threatMetrixTemplate": "some text",
      }
    }
    auth={
      Object {
        "isAuthenticated": false,
        "isFetching": false,
        "loginError": null,
        "menuActive": false,
        "newToken": null,
        "signupError": null,
        "tokenAutoRenew": false,
        "tokenTermsRenew": null,
      }
    }
    cartUpdateSuccess={[Function]}
    checkTokenExpiration={[Function]}
    className="someChild"
    clearApplicationStorage={[Function]}
    clearNotification={[Function]}
    getProspect={[Function]}
    getThreatMetrixTags={[Function]}
    location={
      Object {
        "query": Object {
          "ppid": "777",
        },
        "search": "?ppid=777",
      }
    }
    logoutUser={[Function]}
    notifications={Array []}
    openDialog={[Function]}
    performCartAction={[Function]}
    refreshToken={[Function]}
    removeAllItemsFromBag={[MockFunction]}
    router={
      Array [
        "shopping-bag",
        "shopping-bag",
      ]
    }
    setProspectId={[Function]}
    shoppingBag={
      Object {
        "accessories": Object {},
        "finalized": false,
        "plan": Object {
          "planCode": 0,
        },
        "test": 123,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    syncBag={[Function]}
    title="myLayout"
    toggleRefreshTokenDialog={[Function]}
    toggleVisibilitySignInForm={[Function]}
    userExperience={
      Object {
        "cart": Object {
          "productIds": Array [],
        },
        "dispatch": undefined,
        "isUpdatingCart": false,
        "notifications": Array [],
        "title": "myLayout",
      }
    }
  >
    <div
      className="someChild"
    />
  </div>
  <Footer />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "some text",
      }
    }
  />
</div>
`;
